build/Makefile.global

```diff
@@ -142,6 +142,12 @@ prof-clean:
 prof-use:
 	CCACHE_DISABLE=1 $(MAKE) PROF_FLAGS=-fprofile-use all
 
+%_arginfo.h: %.stub.php
+	if type php >/dev/null 2>/dev/null; then php scripts/dev/gen_stub.php $<; fi
+
+# As we don't track includes, this is just a heuristic
+%.c: %_arginfo.h
+	touch $@
```

Comment:
not sure what the comment means, but i have idea to use compiler dependency analyze:

```makefile
dep depend: .depend

.depend:
    $(CC) -MM source1.c source2.c > $@.tmp
    mv $@.tmp $@

ifeq ($(wildcard .depend),.depend)
include .depend
endif
```

Reply:
One issue here is that this particular Makefile can be in the end run on BSD systems where there might be some make derivative used. Last time I've checked the wildcard and many of these nice features of GNU Make can't be used there. Probably those that build PHP on such system are used to install GNU make and use `gmake` instead but just in case if there is option to go more into Posix style that would be good.